frontend/public/components/catalog-items.jsx

```diff
@@ -25,24 +28,61 @@ export class CatalogTileViewPage extends React.Component {
       allCategories,
       currentCategories: allCategories,
       numItems: _.first(allCategories).numItems,
+      filters: props.filters,
+      activeTabs: ['all'],
+      showAllItemsForCategory: null,
     };
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    const { filters, activeTabs } = this.state;
+
+    if (filters !== prevState.filters) {
+      const allCategories = categorizeItems(this.filterItems());
+
+      if (_.first(activeTabs) === 'all') {
+        this.setState({
+          allCategories,
+          currentCategories: allCategories,
+          numItems: _.first(allCategories).numItems,
+        });
+        return;
+      }
+
+      const mainCategory = _.find(allCategories, { id: _.first(activeTabs) });
+      const curCategories = mainCategory.subcategories || allCategories;
+      const numItems = activeTabs.length < 2 ? mainCategory.numItems : _.find(curCategories, { id: _.last(activeTabs) }).numItems;
```

Comment:
I'd suggest breaking this up into multiple lines to make this easier to read.

```js
const activeCategoryID = _.last(activeTabs);
const activeCategory = activeTabs.length < 2 ? mainCategory : _.find(curCategories, { id: activeCategoryID });
const { numItems } = activeCategory;
```

We might consider keeping a map of categories by id if we have logic like this in several places.